static/app/views/routeAnalyticsContextProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import React, {createContext} from 'react';
+
+import HookStore from 'sentry/stores/hookStore';
+import {Organization} from 'sentry/types';
+
+const DEFAULT_CONTEXT = {
+  setDisableRouteAnalytics: () => {},
+  setRouteAnalyticsParams: () => {},
+  setOrganization: () => {},
+};
+
+export const RouteAnalyticsContext = createContext<{
+  setDisableRouteAnalytics: (disable: boolean) => void;
+  setOrganization: (organization: Organization) => void;
+  setRouteAnalyticsParams: (params: Record<string, any>) => void;
+}>(DEFAULT_CONTEXT);
+
+export default function RouteAnalyticsContextProvider(props: {
+  children: React.ReactNode;
+}) {
+  const useRouteActivatedHook = HookStore.get('react-hook:route-activated')[0];
+  const {setDisableRouteAnalytics, setRouteAnalyticsParams, setOrganization} =
+    useRouteActivatedHook?.(props) || DEFAULT_CONTEXT;
+
+  return (
+    <RouteAnalyticsContext.Provider
+      value={{
+        setDisableRouteAnalytics,
+        setRouteAnalyticsParams,
+        setOrganization,
+      }}
+    >
+      {props.children}
+    </RouteAnalyticsContext.Provider>
+  );
+}
```

Comment:
I think you don't need a context provider here, this can just BE the `useRouteAnalytics` thing I mentioned earlier, you can just get the route props with like `useParams` or whatever

Reply:
@evanpurkhiser Hmmm how do I pass the provided values to the children then? `props.children` isn't a function here. Not sure how to do that except via a provider. This hook is also different than the other hook `useRouteAnalytics` which doesn't have a dependency on getsentry.

Reply:
what do you mean? the values are just returned from the hook?